src/classes/build-error.ts

```diff
@@ -0,0 +1,9 @@
+export class BuildError extends Error {
```

Comment:
We can classify this further into different build type errors, but this is really good to have generic build type error.